compiler/shared_ast/definitions.ml

```diff
@@ -45,6 +45,67 @@ module SubScopeName = Uid.Gen ()
 module StateName = Uid.Gen ()
 
 (** {1 Abstract syntax tree} *)
+module DesugaredVarName : sig
```

Comment:
The `Definitions` module is expected to contain only definitions, which morally allows us to ① have no `.mli` and ② to `open` it throughout.

I know in this case it's harmless since you took care to provide a signature — and also that we're already cheating with this rule a bit with the `Uid.Gen ()` applications — but since the type is not used below in the module, there should be no trouble in putting it in its own file ?

Also:
- why not just `DesugaredVar` ?
- shouldn't this replace `Desugared.Ast.ScopeDef` which is made redundant and could probably be removed ?